ktor-client/ktor-client-cio/common/src/io/ktor/client/engine/cio/Endpoint.kt

```diff
@@ -30,45 +31,52 @@ internal class Endpoint(
 ) : CoroutineScope, Closeable {
     private val address = NetworkAddress(host, port)
 
+    private val mutex = Mutex()
     private val connections: AtomicInt = atomic(0)
-    private val tasks: Channel<RequestTask> = Channel(Channel.UNLIMITED)
     private val deliveryPoint: Channel<RequestTask> = Channel()
     private val maxEndpointIdleTime: Long = 2 * config.endpoint.connectTimeout
 
-    private val postman = launch {
+    val lastActivity = atomic(GMTDate())
+
+    private val timeout = launch(coroutineContext + CoroutineName("Endpoint timeout($host:$port")) {
```

Comment:
Missing close bracket in the coroutine name pattern after the port substitution

Reply:
Btw, is it safe to put a hostname into coroutine name and to thread name that sometimes goes to some native stuff and may have some effects we can't trace